### PR TITLE
User onboarding flow and New channel tour

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
This PR introduces a UI component for the new tutorial tour tip component that isn't dependent on react-bootstrap and Use tippy instead to make it useable in our other products like Boards, Playbooks.

A new on boarding channels tour which will be replacing the existing one for all users and simple tour for other tasks( system console, user profile, invite modal)

#### Ticket Link
https://community-daily.mattermost.com/boards/workspace/fpbb7orqpiyfxqgg6fcu91y5xo/b6ropkdiowj8eugbt8kifrew79h/vsha9cczxb3bofripyh3o549s6r/c73y498ooftd65f7uczqbxouuhh

#### Related Pull Requests
``NONE``

#### Release Note
```release-note
* UI: New tutorial tour tip component introduction (use new library React-tippy)
```

### CODE REVIEW

**Comment by AshishDhama** [comment]
@matthewbirtch except navigation via dots in tour and check have fixed all the issue you have mentioned in https://github.com/mattermost/mattermost-webapp/pull/9774.

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
I'm not sure if the below issues belong to this PR or to Nevy's, but let me know.

1. Seems like there is an issue with the task list when switching between boards and channels. Things are shifting around


2. The plus button that the menu originates from should be highlighted/cutout in this step - if we can figure that out
image

3. Somehow I got this screen, but I only completed the first 2 tasks. It seemed to happen after clicking a previously completed task
image

**Review by esethna — APPROVED** [review]
This is coming along great! Approving so I don't block this PR, but I noticed a few things:

1. The profile menu was greyed out when I clicked "Complete your profile" in the task list

image

2. I assume the logic for marking the tasks complete hasn't been completed since they are being marked complete immediately after clicking them?

3. The download apps task list item didn't do anything

4. Hover effects seem missing when the menu items are open (same the complete profile task list item when clicked): https://gyazo.com/776594d3eca292e417b909be5e301a63

**Comment by neallred** [comment]
Code generally looks good to me; finishing up the rest of my review and post those comments a bit later. I wanted to call out sooner what I think is the main issue we need to resolve for this PR. The existing tour is stored in the preferences table as:

```
           userid           |            category            |            name            |     value
----------------------------+--------------------------------+----------------------------+---------------
 gan834poyibzjrodff577q3aee | tutorial_step                  | gan834poyibzjrodff577q3aee | 2
 gan834poyibzjrodff577q3aee | tutorial_step_auto_tour_status | gan834poyibzjrodff577q3aee | 0
```

This PR uses a new set of code for the tours, which seems fine to me. That new code also adds and reads from a new preferences entry, i.e.

```
           userid           |            category            |            name            |     value
----------------------------+--------------------------------+----------------------------+---------------
 gan834poyibzjrodff577q3aee | channels_tour                  | auto_tour_status           | 0
 gan834poyibzjrodff577q3aee | channels_tour                  | on_boarding_step           | 4
```

It uses the new entry, and disregards the old one. While this is not an issue for new users, is does mean all existing users (admin or otherwise) get put through a tour again, which is pretty similar to the existing, in-prod/deployed one (the content isn't exactly the same). @AshishDhama  @esethna @matthewbirtch are we ok with that? If not, if we're going to add the new `on_boarding_step` entry to the table, we should add some sort of code to migrate over existing user progress. Not sure if all steps have a tranlsation between the old tour set and the new one. If they don't, we could at least code it in so that users that were `999`/`FINISHED` before in the old tour preference keep that status in the new `on_boarding_step` tour preference.

**Comment by AshishDhama** [comment]
@neallred Thanks for pointing this out, Old Users will not get this tour as there is additional conditions I am going to add, that’s what I wanted to make sure when I asked about ``current onboarding flow`` next steps one I showed you in our 1-1.
This tour want be available as default, there is trigger condition which needs to be satisfied to start it. Will push soon.

**Comment by neallred** [comment]
Cool, will take a look when that's pushed. Thanks!

**Comment by AshishDhama** [comment]
@matthewbirtch Thanks for such a detailed review,

> 6. For the video modal, it starts out blurred. How come?
could you please share image of this, I am not sure as this is Iframe it might be that’s how wista do it

> 7. Also, for the video modal, can we add the standard modal rounded corner (assume that's a variable?). There is no rounded corner right now.
It was like that in Figma, will add border-radius

> 17. We need the task list to follow us even while we're in the System Console.
I checked when you mentioned about it on MM, Due to use of styled components it wasn’t rendering in admin console(after explicitly putting it there), Then Nevy dropped a message that she will fix it, so I dropped further investigation I am not sure if we can do that with styled components but will check and get back to you tomorrow with answers.

**Comment by AshishDhama** [comment]
@matthewbirtch Thanks for such a detailed review,

> 6. For the video modal, it starts out blurred. How come?
could you please share image of this, I am not sure as this is Iframe it might be that’s how wista do it

> 7. Also, for the video modal, can we add the standard modal rounded corner (assume that's a variable?). There is no rounded corner right now.
It was like that in Figma, will add border-radius

> 17. We need the task list to follow us even while we're in the System Console.
I checked when you mentioned about it on MM, Due to use of styled components it wasn’t rendering in admin console(after explicitly putting it there), Then Nevy dropped a message that she will fix it, so I dropped further investigation I am not sure if we can do that with styled components but will check and get back to you tomorrow with answers.

**Comment by matthewbirtch** [comment]
> could you please share image of this, I am not sure as this is Iframe it might be that’s how wista do it

This is what I see when it opens

**Review by esethna — CHANGES_REQUESTED** [review]
Looking great @AshishDhama. Few things in addition to what Matt has already mentioned:

1. The sound on the video wasn't working for me 
2. When are we removing the previous "getting started" stuff in the sidebar so it doesn't conflict with the new task list?

File: components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx
Comment (AshishDhama) [reply components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx:174]: We don’t need to skip this tip as default channel (town) will be fall back

File: components/global_onboarding_list/onboarding_checklist.tsx
Comment (AshishDhama) [reply components/global_onboarding_list/onboarding_checklist.tsx:38]: Thanks for the suggestion

**Comment by AshishDhama** [comment]
@esethna 
> 1. The sound on the video wasn't working for me

 Not sure about 1 as we don’t have any control over video using Wista embedded in Iframe

> 2. When are we removing the previous "getting started" stuff in the sidebar so it doesn't conflict with the new task list?
 
We aren’t removing them for now except for first admin user and for first admin user it will be done once we merge all branches

File: components/onboarding_tasks/constants.ts
Comment (neallred) [inline components/onboarding_tasks/constants.ts:23]: Nice changes! That is a lot of renames. Could we also change these last string values to `onboarding`? Looks like there's one left in `utils/constants.jsx`, too.

Comment (AshishDhama) [reply components/onboarding_tasks/constants.ts:23]: Yeah sure

**Comment by esethna** [comment]
@Willyfrog please help review to unblock merging

**Review by neallred — APPROVED** [review]
Looks good to me now!

**Comment by matthewbirtch** [comment]
@AshishDhama can you have a look at the following?

1. Still seeing this: When I clicked on 'Download the desktop and mobile apps', it opened and new tab and when I returned to original tab, the task list was gone. Can it stay open in this case?

2. When I click on 'Visit the System Console...' task, I saw this the first time. It only happened the first time though it seems
image

3. You're already aware of this one, but adding it to the list. When the task list if open in the system console, the header area is highlighted
image

I think that's it!

**Comment by esethna** [comment]
Adding the QA deferred label as we will test all the PRs together after merge

cc// @jgilliam17 @Willyfrog

**Review by esethna — APPROVED** [review]
Nice work @AshishDhama!

**Review by Willyfrog — APPROVED** [review]

File: components/widgets/tour_tip/tour_tip_backdrop.tsx
Comment (koox00) [inline components/widgets/tour_tip/tour_tip_backdrop.tsx:66]: could be memo-ed but 0/5 on this.

File: i18n/en.json
Comment (koox00) [inline i18n/en.json:3836]: I don't get why we are changing the ids, It seems that the strings are the same.
Won't we lose our already translated  ones this way?

File: components/onboarding_tasklist/onboarding_tasklist.tsx
Comment (AshishDhama) [reply components/onboarding_tasklist/onboarding_tasklist.tsx:246]: Those id are also new we merged that pr this week only, so just made all off them inline.

File: i18n/en.json
Comment (AshishDhama) [reply i18n/en.json:3836]: these aren’t old Id's, were created as part of growth spike so to keep naming inline changed them.

File: components/onboarding_tour/utils.ts
Comment (AshishDhama) [reply components/onboarding_tour/utils.ts:39]: Because we have used -1 in previous implementation so want keep that door open

File: components/onboarding_tour/invite_people_tour_tip.tsx
Comment (AshishDhama) [reply components/onboarding_tour/invite_people_tour_tip.tsx:35]: One time use only

File: components/onboarding_tour/create_and_join_channels_tour_tip.tsx
Comment (AshishDhama) [reply components/onboarding_tour/create_and_join_channels_tour_tip.tsx:35]: same as at other places one time use

File: components/onboarding_tour/customize_your_experience_tour_tip.tsx
Comment (AshishDhama) [reply components/onboarding_tour/customize_your_experience_tour_tip.tsx:38]: same as at other places one time use

File: components/onboarding_tasklist/onboarding_tasklist.tsx
Comment (AshishDhama) [reply components/onboarding_tasklist/onboarding_tasklist.tsx]: removed this logic.

File: components/onboarding_tasks/onboarding_tasks_manager.tsx
Comment (AshishDhama) [reply components/onboarding_tasks/onboarding_tasks_manager.tsx:139]: savePreferences is required as that’s a local func, Sorry for mixing up the names changing that now

File: components/onboarding_tasks/onboarding_tasks_manager.tsx
Comment (AshishDhama) [reply components/onboarding_tasks/onboarding_tasks_manager.tsx:133]: name mix up

**Review by matthewbirtch — APPROVED** [review]
Thanks for resolve those last few issues @AshishDhama. Looks great!

**Review by neallred — APPROVED** [review]

File: components/root/root.jsx
Comment (AshishDhama) [reply components/root/root.jsx:444]: Don't have answer but will check but I think this isn’t a blocker.

**Comment by esethna** [comment]
@koox00 can you give this another review asap as we are aiming to merge this as soon as we can

**Review by koox00 — APPROVED** [review]
💪

**Comment by koox00** [comment]
@esethna done. @AshishDhama there are conflicts to be resolved though

**Pull request merged**
